### ISSUE TITLE
Right click menu has black corners

### ISSUE THREAD

**Issue by BeeverFeever** [nuclearcodecat/shimmer#8]
As per my last issue, I can't get a screenshot of it as it closes before the screenshot is taken. Pretty much as the title says, the right click menu has pure black corners. I have a feeling this may have something to do with my window manager/compositor however I'm really not sure.

**Comment by nuclearcodecat** [comment]
you'll have to send me a screenshot. it's possible to disable popup autohide in the browser toolbox, but if you never set it up, i've read that there is a method `ui.popup.disable_autohide` in `about:config`. 

https://superuser.com/questions/1465087/how-to-analyze-popup-window-with-inspector-on-firefox#1465104

**Comment by BeeverFeever** [comment]
Oh ok, didn't know that was a thing. I'll give it a shot now.

**Comment by BeeverFeever** [comment]
![image](https://github.com/user-attachments/assets/9f7e843b-e50d-4d18-bb98-14d65831e237)

**Comment by nuclearcodecat** [comment]
thank you :3. i'll take a look at the code later. this might be fixed by hiding an overflow a the right element or moving `border-radius` to a higher-level element

**Comment by BeeverFeever** [comment]
Ok, I'm having a play around with the code now so I will see if I can come up with anything.

**Comment by nuclearcodecat** [comment]
did you try this already?

```
.menupopup-arrowscrollbox {
  overflow: hidden !important;
}
```

**Comment by BeeverFeever** [comment]
Yeah that doesnt work, I also tried moving border-radius to there as well and that doesn't work.

**Comment by nuclearcodecat** [comment]
weird, try
```
.menupopup-arrowscrollbox {
  background: rgba(0, 0, 0, 0) !important;
}
```

**Comment by BeeverFeever** [comment]
![image](https://github.com/user-attachments/assets/d10ceb2a-ab6d-458b-91fd-baec213af14d)

Thats what happens.

**Comment by nuclearcodecat** [comment]
so it must be something even higher in the dom, the only thing that really could affect this (on my pc) is `#contentAreaContextMenu`, you could try the background thing here, i don't really have any other ideas

**Comment by BeeverFeever** [comment]
Nope, changing that doesn't do anything either. I may have a look at some other userchromes to see how they do it. I can't find anything else either.

**Comment by nuclearcodecat** [comment]
keep me updated :). maybe i'll boot up a vm to try some things later

**Comment by BeeverFeever** [comment]
Yeah will do.

**Comment by BeeverFeever** [comment]
Is there a way I can test the userchrome with a bone stock firefox, on my system (not a vm). Just to rule out other things affecting it?

**Comment by nuclearcodecat** [comment]
i don't really know :(. please tell me what system and window manager you are using

**Comment by BeeverFeever** [comment]
Dang ok. Im on arch with i3wm and firefox 132.0

**Comment by BeeverFeever** [comment]
I solved the problem. It was compositor related. I used to not use a compositor at all and I ended up just starting picom which in turn solved the issue (with the correct config for picom that is). My guess is that firefox uses a native/os/window manager popup menu for the popups like the right click menu and bookmarks menu. This means that firefox doesn't have complete control over the look of said menu's.

In short, problem solved by running picom with correct config.
